src/address.rs

```diff
@@ -1,8 +1,16 @@
+use halo2::pasta::pallas;
+
 use crate::keys::Diversifier;
 
 /// A shielded payment address.
 #[derive(Debug)]
 pub struct Address {
     d: Diversifier,
-    pk_d: (),
+    pk_d: pallas::Point,
```

Comment:
`pallas::Point` is used in several places for different purposes in `orchard`, should we have a wrapper type like `orchard::keys::DiversifiedTransmissionKey` or whatever?  That's what the spec calls pk_d on the March 4 version.